cpp/src/neighbors/cagra_c.cpp

```diff
@@ -83,6 +83,34 @@ void* _build(cuvsResources_t res, cuvsCagraIndexParams params, DLManagedTensor*
   return index;
 }
 
+template <typename T>
+void _extend(cuvsResources_t res,
+             cuvsCagraExtendParams params,
+             cuvsCagraIndex index,
+             DLManagedTensor* additional_dataset_tensor)
+{
+  auto dataset   = additional_dataset_tensor->dl_tensor;
+  auto index_ptr = reinterpret_cast<cuvs::neighbors::cagra::index<T, uint32_t>*>(index.addr);
+  auto res_ptr   = reinterpret_cast<raft::resources*>(res);
+
+  auto extend_params           = cuvs::neighbors::cagra::extend_params();
+  extend_params.max_chunk_size = params.max_chunk_size;
```

Comment:
This doesn't need to be changed in this PR (especially since the rest of cagra/ivf-* doesn't follow this right now) - but I think we should start re-using the C-api parameter structs in the C++ api , like we are doing in the distances here https://github.com/rapidsai/cuvs/blob/72154b0b806c106300b52870f6113fdda3f87f0b/cpp/include/cuvs/distance/distance.hpp#L27. This means we don't have to have two different identical structs in the C and C++ api, and copy between them you are doing here

Reply:
+1 to this. It increases the maintenance burden when we have parallel objects like this.

Reply:
@ajit283 as Ben mentioned, it doesn't have to be changed in this PR, but can you create a Github issue for this and reference the issue number in a TODO here in the code? This way we won't lose sight of it.

Reply:
One problem that arises when we do this is that we have to include the C header file in the C++ header file. While it works for distances enum, here the issue is that we also have dlpack headers which we don't link to the C++ target.

So we'll have to pull out all the structs into another file to make it work.

@cjnolet @benfred